src/TLuaInterpreter.cpp

```diff
@@ -7989,6 +8033,71 @@ int TLuaInterpreter::isActive(lua_State* L)
     return 1;
 }
 
+// Documentation: https://wiki.mudlet.org/w/Manual:Lua_Functions#isActiveID
+int TLuaInterpreter::isActiveID(lua_State* L)
+{
+    auto id = getVerifiedInt(L, __func__, 1, "item ID");
+    if (id < 0) {
+        // Must be zero or more but doesn't seem to be:
+        return warnArgumentValue(L, __func__, "item ID as %d does not seem to be a positive integer", id);
+    }
+
+    // Although we only use 4 ASCII strings the user may not enter a purely
+    // ASCII value which we might have to report...
+    QString type = getVerifiedString(L, __func__, 2, "item type");
+
+    Host& host = getHostFromLua(L);
+    if (!type.compare(QLatin1String("timer"), Qt::CaseInsensitive)) {
+        auto pT = host.getTimerUnit()->getTimer(id);
+        if (!pT) {
+            return warnArgumentValue(L, __func__, "timer ID %d does not exist", id);
+        }
+        lua_pushboolean(L, pT->isActive());
+        return 1;
+    }
+    if (!type.compare(QLatin1String("trigger"), Qt::CaseInsensitive)) {
+        auto pT = host.getTriggerUnit()->getTrigger(id);
+        if (!pT) {
+            return warnArgumentValue(L, __func__, "trigger ID %d does not exist", id);
+        }
+        lua_pushboolean(L, pT->isActive());
+        return 1;
+    }
+    if (!type.compare(QLatin1String("alias"), Qt::CaseInsensitive)) {
+        auto pT = host.getAliasUnit()->getAlias(id);
+        if (!pT) {
+            return warnArgumentValue(L, __func__, "alias ID %d does not exist", id);
+        }
+        lua_pushboolean(L, pT->isActive());
+        return 1;
+    }
+    if (!type.compare(QLatin1String("keybind"), Qt::CaseInsensitive)) {
+        auto pT = host.getKeyUnit()->getKey(id);
+        if (!pT) {
+            return warnArgumentValue(L, __func__, "key-binding ID %d does not exist", id);
+        }
+        lua_pushboolean(L, pT->isActive());
+        return 1;
+    }
+    if (!type.compare(QLatin1String("button"), Qt::CaseInsensitive)) {
+        auto pT = host.getActionUnit()->getAction(id);
+        if (!pT) {
+            return warnArgumentValue(L, __func__, "button/menu/toolbar ID %d does not exist", id);
+        }
+        lua_pushboolean(L, pT->isActive());
+        return 1;
+    }
+    if (!type.compare(QLatin1String("script"), Qt::CaseInsensitive)) {
+        auto pT = host.getScriptUnit()->getScript(id);
+        if (!pT) {
+            return warnArgumentValue(L, __func__, "script ID %d does not exist", id);
+        }
+        lua_pushboolean(L, pT->isActive());
+        return 1;
+    }
+    return warnArgumentValue(L, __func__, qsl("invalid item type '%1' given, it should be one (case insensitive) of: 'alias', 'button', 'script', 'keybind', 'timer' or 'trigger'").arg(type));
```

Comment:
Improve readability and sentence flow
```suggestion
    return warnArgumentValue(L, __func__, qsl("invalid item type '%1' given, it should be one of: 'alias', 'button', 'script', 'keybind', 'timer' or 'trigger' (case insensitive)").arg(type));
```

Reply:
Strangely I feel the original (copied from the existing function IIRC) actually reads better IMHO. Oh, only the `isActive(...)` reads like that - the existing `exists(...)` one erroneously omits the `(case insensitive)` altogether - so I am minded to leave it like I had it then if we want to change it we can do all three instances and insert it in for the fourth.